src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -823,6 +823,12 @@ private function addSerializerSection(ArrayNodeDefinition $rootNode)
                                 ->end()
                             ->end()
                         ->end()
+                        ->arrayNode('default_context')
+                            ->normalizeKeys(false)
+                            ->useAttributeAsKey('name')
+                            ->defaultValue([])
+                            ->prototype('variable')->end()
+                        ->end()
```

Comment:
This doesn't allow for things like the `circular_reference_handler`. I mean, you can set it, but there's no way to set this to a service, for example - none of the handler stuff will work. We need to check for the handler options explicitly in `FrameworkExtension` and convert them to References. In the docs, I guess we'll need to document these handlers as services that should use the `__invoke()` method.

Reply:
Maybe could be also do something more generic: check if a service exists corresponding to a given string exists, if it's the case, convert it to a reference. WDYT?